examples/lib/stories/camera_and_viewport/zoom.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/gestures.dart';
+
+class ZoomGame extends BaseGame with ScrollDetector, ScaleDetector {
+  final Vector2 viewportResolution;
+  late SpriteComponent flame;
+
+  Vector2? lastDelta;
+
+  ZoomGame({
+    required this.viewportResolution,
+  });
+
+  @override
+  Future<void> onLoad() async {
+    final flameSprite = await loadSprite('flame.png');
+
+    viewport = FixedResolutionViewport(viewportResolution);
+
+    final flameSize = Vector2(149, 211);
+    add(
+      flame = SpriteComponent(
+        sprite: flameSprite,
+        size: flameSize,
+      )..anchor = Anchor.center,
+    );
+    camera.followComponent(
+      flame,
+      relativeOffset: Anchor.center.toVector2(),
+    );
+  }
+
+  static const zoomPerScrollUnit = 0.001;
+  @override
+  void onScroll(PointerScrollInfo event) {
+    camera.zoom += event.scrollDelta.game.y * zoomPerScrollUnit;
+  }
+
+  @override
+  void onScaleUpdate(ScaleUpdateInfo info) {
+    final scale = lastDelta;
+    if (scale != null) {
+      final delta = info.scale.game - scale;
```

Comment:
why are you subtracting the lastDelta here? can't you just inc/dec the zoom by the amount scaled?

Reply:
I have actually choose a very bad name for this variable.

This isn't the lastDelta but rather the lastScale.

This event don't have a delta, the scale is just the current average distance from each focal touch point.

So to have the delta we need to keep track of the last scale.

It took me a while to understand that and I forgot to rename the variable, I will rename it so the code makes more sense.